From 1c5c10b4d766646c6eed75c2105d8d6180175dd8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Verschelde?= <rverschelde@gmail.com>
Date: Fri, 20 Oct 2017 18:40:41 +0200
Subject: [PATCH] Mono: Fix static linking on macOS

---
 modules/mono/config.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/modules/mono/config.py b/modules/mono/config.py
index 13b9a4b1e..056a1d81b 100644
--- a/modules/mono/config.py
+++ b/modules/mono/config.py
@@ -125,7 +125,11 @@ def configure(env):
             else:
                 env.Append(LIBS=[mono_lib])
 
-            env.Append(LIBS=['m', 'rt', 'dl', 'pthread'])
+            if sys.platform == "darwin":
+                env.Append(LIBS=['pthread'])
+            elif sys.platform == "linux" or sys.platform == "linux2":
+                env.Append(LIBS=['m', 'rt', 'dl', 'pthread'])
+
         else:
             if mono_static:
                 raise RuntimeError('mono-static: Not supported with pkg-config. Specify a mono prefix manually')
-- 
2.13.5

